doc/how_to/concurrency/load_balancing.md

```diff
@@ -1,12 +1,156 @@
+Setting up load balancing is a huge topic, dependent on the precise system you
+are using, so the example below will be very generic. In most cases
+you set up a reverse proxy (like NGINX) to distribute the load across multiple
+application servers. If you are using a system like Kubernetes, it will also
+handle spinning up and shutting down the servers for you, and can even do so dynamically, depending
+on the amount of concurrent users to ensure that you are not wasting resources
+when there are fewer users, and / or when there is less computing demand.
 
 <figure>
-<img src="https://www.nginx.com/wp-content/uploads/2014/07/what-is-load-balancing-diagram-NGINX-1024x518.png" width="768"></img>
+<img src="https://assets.holoviz.org/panel/how_to/concurrency/what-is-load-balancing-diagram-NGINX.png" width="768"></img>
 <figcaption>Diagram showing concept of load balancing (NGINX)</figcaption>
 </figure>
 
-Load balancing is the most complex approach to set up but is guaranteed to improve concurrent usage of your application since different users are not contending for access to the same process or even necessarily the same physical compute and memory resources. At the same time it is more wasteful of resources since it potentially occupies multiple machines and since each process is isolated there is no sharing of cached data or global state.
+Load balancing is the most complex approach to set up but is guaranteed to
+improve concurrent usage of your application since different users are not
+contending for access to the same process or even necessarily the same physical
+compute and memory resources. At the same time it is more wasteful of resources
+since it potentially occupies multiple machines and since each process is
+isolated there is no sharing of cached data or global state.
 
-To get started configuring a load balancer take a look at the [Bokeh documentation](https://docs.bokeh.org/en/latest/docs/user_guide/server/deploy.html#load-balancing) which provides example configurations for Apache and NGINX.
+To get started configuring a load balancer take a look at the [Bokeh
+documentation](https://docs.bokeh.org/en/latest/docs/user_guide/server/deploy.html#load-balancing).
+
+## Use NGINX and Containers with Panel along with other Bokeh extensions
+
+Panel is built on top of Bokeh and uses the Bokeh server to serve applications. To serve Panel-specific resources, Panel is defined as a Bokeh extension.
+Panel's resources are available on a CDN, but if you use other Bokeh Extensions, serving them directly from the Bokeh server can be beneficial.
```

Comment:
Maybe expand a little on why it can be beneficial.

Reply:
```suggestion
Panel's resources are available on a CDN, but if you use other Bokeh Extensions, serving them directly from the Bokeh server can be the only way to access them. 
```

Reply:
> Is it used as an example of a custom local extension that wouldn't be available on a CDN

Yes. Or something where you can't change the resources to get pulled from a CDN. 

> I'm trying to understand the consequences of all of this.

 If you haven't done that already, I suggest reading the original issue.

Reply:
Perfect thanks!